libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -123,7 +123,7 @@ void NavEKF2_core::setAidingMode()
             stateStruct.position.z = -meaHgtAtTakeOff;
         } else if (frontend->_fusionModeGPS == 3) {
             // We have commenced aiding, but GPS useage has been prohibited so use optical flow only
-            hal.console->printf("EKF2 is using optical flow\n");
+            hal.console->printf("EKF2 IMU%i is using optical flow\n",imu_index);
```

Comment:
this should be IMU%u and cast imu_index to unsigned, like this:
  hal.console->printf("EKF2 IMU%u is using optical flow\n", (unsigned)imu_index);
what you've done will work, but isn't as portable